keras_cv/layers/preprocessing/random_rotation.py

```diff
@@ -0,0 +1,220 @@
+
+    def __init__(
+        self,
+        factor,
+        fill_mode="reflect",
+        interpolation="bilinear",
+        seed=None,
+        fill_value=0.0,
+        bounding_box_format=None,
+        **kwargs,
+    ):
+        super().__init__(seed=seed, force_generator=True, **kwargs)
+        self.factor = factor
+        if isinstance(factor, (tuple, list)):
+            self.lower = factor[0]
+            self.upper = factor[1]
+        else:
+            self.lower = -factor
+            self.upper = factor
+        if self.upper < self.lower:
+            raise ValueError(
+                "Factor cannot have negative values, " "got {}".format(factor)
+            )
+        preprocessing.check_fill_mode_and_interpolation(fill_mode, interpolation)
+        self.fill_mode = fill_mode
+        self.fill_value = fill_value
+        self.interpolation = interpolation
+        self.seed = seed
+        self.bounding_box_format = bounding_box_format
+
+    def get_random_transformation(self, image=None, label=None, bounding_box=None):
+        min_angle = self.lower * 2.0 * np.pi
+        max_angle = self.upper * 2.0 * np.pi
+        angle = self._random_generator.random_uniform(
+            shape=[1], minval=min_angle, maxval=max_angle
+        )
+        return {"angle": angle}
+
+    def augment_image(self, image, transformation):
+        image = preprocessing.ensure_tensor(image, self.compute_dtype)
+        original_shape = image.shape
+        image = tf.expand_dims(image, 0)
+        image_shape = tf.shape(image)
+        img_hd = tf.cast(image_shape[H_AXIS], tf.float32)
+        img_wd = tf.cast(image_shape[W_AXIS], tf.float32)
+        angle = transformation["angle"]
+        output = preprocessing.transform(
+            image,
+            preprocessing.get_rotation_matrix(angle, img_hd, img_wd),
+            fill_mode=self.fill_mode,
+            fill_value=self.fill_value,
+            interpolation=self.interpolation,
+        )
+        output = tf.squeeze(output, 0)
+        output.set_shape(original_shape)
+        return output
+
+    def augment_bounding_boxes(self, image, bounding_boxes, transformation):
+        bbox_dtype = bounding_boxes.dtype
+        if self.bounding_box_format is None:
+            raise ValueError(
+                "bounding_box_format has not been specified during layer "
+                "initialization. Please refer docstring for more information."
+            )
+        else:
+            bounding_boxes = bounding_box.convert_format(
+                bounding_boxes, source=self.bounding_box_format, target="xyxy"
+            )
+        image = tf.expand_dims(image, 0)
+        image_shape = tf.shape(image)
+        h = image_shape[H_AXIS]
+        w = image_shape[W_AXIS]
+        # origin coordinates, all the points on the image are rotated around
+        # this point
+        origin_x, origin_y = int(h / 2), int(w / 2)
+        angle = transformation["angle"]
+        angle = -angle
```

Comment:
Any reason we are doing a negation here?

Reply:
negating the angle makes sure the rotation is anti clockwise.

Reply:
i see. Since I didn't find the negation part in image, we probably should document it here.